### PR TITLE
Throw exception on invalid API resource endpoint return,

### LINKED ISSUE #303: Unable to create new entity: Cannot read property 'id' of null
I create a new entity and put it to a repeat block:
```html
<tr repeat.for="client of clients">
	<td>${client.id}</td>
	<td><input type="text" value.bind="client.name"></td>
	<td>${client.is_default}</td>
</tr>
```

```javascript
newClient() {
    this.clients.push(this.em.getEntity(Client));
}
```
But when I want to save the client:
```javascript
saveClients() {
	for (let client of this.clients) {
		client.save();
	}
}
```

I get the following error:
```
Uncaught (in promise) TypeError: Cannot read property 'id' of null
    at eval (aurelia-orm.js:396)
    at <anonymous>
```
```
395: return this.getTransport().create(this.getResource(), this.asObject(true)).then(function (created) {
396:        _this4.setId(created[_this4.getIdProperty()]);
397:        response = created;
```
The entity meta data has the property ``idProperty:"id"``

### PR DESCRIPTION
**Pull request by mreiche**
That fixes https://github.com/SpoonX/aurelia-orm/issues/303

### CODE REVIEW

File: src/entity.js [hunk 1]
```diff
@@ -139,6 +139,7 @@ export class Entity {
     return this.getTransport()
       .create(this.getResource(), this.asObject(true))
       .then(created => {
+        if (created===null||created[this.getIdProperty()]===undefined) throw new Error('The API resource did not return the id property "'+this.getIdProperty()+'"');
```
Comment (RWOverdijk) [inline src/entity.js]: Please follow the code conventions. Spaces surrounding or statement and curly braces.

**Review by RWOverdijk — CHANGES_REQUESTED** [review]
Could you update the tests and change your commit messages according to the [guidelines](https://github.com/SpoonX/about/blob/master/CONTRIBUTING.md#commit)?

File: src/entity.js (hunk 1 above)
Comment (mreiche) [reply src/entity.js]: Of course. Sorry for that.

**Comment by mreiche** [comment]
Ok, I'm going to fix the tests now seriously.

**Comment by mreiche** [comment]
Hmm.. I have no glue how to implement that "positive false" test for the exception. I'm not really familiar with Javascript/Karma. Where are the services mocked? Any tips and hints?

**Comment by RWOverdijk** [comment]
You changed the error message, so the check for the error message needs to be different. It's in the test directory.

Re: commit messages, They're still not correct. Use the imperative, present tense. Also, the format is: `type(scope): message`. Example:

`test(repository): update exception message asserts`

**Comment by mreiche** [comment]
> You changed the error message, so the check for the error message needs to be different. It's in the test directory.

This is not the only problem. I fixed that locally. But I'm unable to avoid:
```
ERROR: 'Potentially unhandled rejection [1] Error: The API resource did not return @idProperty for entity.
    at eval (http://localhost:9876/base/src/entity.js:174:15)
    at tryCatchReject (http://localhost:9876/base/jspm_packages/system-polyfills.src.js?9e8733cdc1a619a0238d6b06eeca4ba4733673d7:1188:30)
    at runContinuation1 (http://localhost:9876/base/jspm_packages/system-polyfills.src.js?9e8733cdc1a619a0238d6b06eeca4ba4733673d7:1147:4)
    at Fulfilled.when (http://localhost:9876/base/jspm_packages/system-polyfills.src.js?9e8733cdc1a619a0238d6b06eeca4ba4733673d7:935:4)
    at Thenable.Pending.run (http://localhost:9876/base/jspm_packages/system-polyfills.src.js?9e8733cdc1a619a0238d6b06eeca4ba4733673d7:826:13)
    at Scheduler._drain (http://localhost:9876/base/jspm_packages/system-polyfills.src.js?9e8733cdc1a619a0238d6b06eeca4ba4733673d7:102:19)
    at Scheduler.drain (http://localhost:9876/base/jspm_packages/system-polyfills.src.js?9e8733cdc1a619a0238d6b06eeca4ba4733673d7:67:9)
    at MutationObserver.run (http://localhost:9876/base/jspm_packages/system-polyfills.src.js?9e8733cdc1a619a0238d6b06eeca4ba4733673d7:273:4)'
Chrome 58.0.3029 (Mac OS X 10.12.5) Entity .save() Should call .create on REST without an ID. (custom entity) FAILED
        Error: Timeout - Async callback was not invoked within timeout specified by jasmine.DEFAULT_TIMEOUT_INTERVAL.
```

**Comment by RWOverdijk** [comment]
@mreiche Do you need my help with this?

**Comment by mreiche** [comment]
> Do you need my help with this?
Yes

**Pull request closed without merging**